cadence/server/public/js/aria.js

```diff
@@ -0,0 +1,197 @@
+var streamSrcURL = "" // this is used by the stream source loader
+
+$(window).on("load", function(e) {
+	getRadioData()
+	connectToSSE()
+	postSearch()
+	setSearchRequestFunctions()
+});
+
+function getRadioData() {
+	$.ajax({
+		type: 'GET',
+		url: "/api/version",
+		dataType: "json",
+		success: function(data) {
+			document.getElementById("release").innerHTML = data.Version;
+		},
+		error: function() {
+			document.getElementById("release").innerHTML = "(N/A)";
+		}
+	});
+	$.ajax({
+		type: 'GET',
+		url: "/api/nowplaying/metadata",
+		dataType: "json",
+		success: function(data) {
+			$('#song').text(data.Title)
+			$('#artist').text(data.Artist)
+		},
+		error: function() {
+			$('#song').text("-")
+			$('#artist').text("-")
+		}
+	});
+	$.ajax({
+		type: 'GET',
+		url: "/api/nowplaying/albumart",
+		dataType: "json",
+		success: function(data) {
+			$('#artwork').attr("src", "data:image/jpeg;base64," + data.Picture);
+		},
+		error: function() {
+			$('#artwork').attr("src", "");
+		}
+	});
+	$.ajax({
+		type: 'GET',
+		url: "/api/listenurl",
+		dataType: "json",
+		success: function(data) {
+			if (data.ListenURL == "-/-") {
+				$('#status').html("Disconnected from server.")
+			} else {
+				streamSrcURL = location.protocol + "//" + data.ListenURL
+				document.getElementById("stream").src = streamSrcURL;
+				$('#status').html("Connected: <a href='"+ streamSrcURL + "'>" + streamSrcURL + "</a>")
+			}
+		},
+		error: function() {
+			document.getElementById("stream").src = "";
+			$('#status').html("Disconnected from server.")
+		}
+	});
+}
+
+var retry = 1;
+
+function connectToSSE() {
+	let eventSource = new EventSource("/api/radiodata/sse");
```

Comment:
I finally get to use modern push APIs when I put this in CadenceBot :D

Reply:
The one drawback I've found is SSE can start to error out after long intervals (100s). I think it has to do with it being deployed onto a K8s cluster. I don't see these issues when deployed onto a local machine through Docker. My bandaid solution is to [reconnect on error](https://github.com/kenellorando/cadence/pull/201/files/78fd04a4c8345b17368da710985dff112e9d75bb#diff-7547750605a8d3d361a1f3a7b8f57f176b3c6507788e7ed3e8505c32e635a714R74-R77).